tests/bpf2c_tests/expected/atomic_instruction_fetch_add_sys.c

```diff
@@ -37,7 +37,7 @@ static NTSTATUS
 _bpf2c_npi_client_detach_provider(_In_ void* client_binding_context);
 
 static const NPI_CLIENT_CHARACTERISTICS _bpf2c_npi_client_characteristics = {
-    0,                                  // Version
+    1,                                  // Version
```

Comment:
Can you update the PR description to explain why a version change is needed?  I couldn't follow it.

Reply:
BPF programs that use global variables require that the address of the map value be preloaded. If it isn't the BPF program will bugcheck the kernel. To prevent this, the version of the NMR contract is bumped to 1 to allow the client to detect older versions of eBPF core and gracefully fail to load (as opposed to crashing).

Reply:
@dthaler should we update the client characteristics version or the NPI version for this case? This is a breaking change in that the client will only work with the newer version of the provider.

My reasoning is as follows:
This *is* a backwards compatible change with respect to the NPI contract as the metadata table is versioned via the size. Older clients can continue to attach and it will work correctly. The problem is that there is a need to prevent newer clients from attaching to older runtimes. Detection based on NPI characteristics seems to meet that requirement.

Reply:
> @dthaler should we update the client characteristics version or the NPI version for this case? This is a breaking change in that the client will only work with the newer version of the provider.

I don't remember offhand but I suspect I answered it in the NMR presentation I gave in 2023, so check that recording :)

Reply:
As per the MSDN this should always be zero. Need to revisit versioning.
https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/netioddk/ns-netioddk-_npi_client_characteristics

Reply:
Ah right, this is the version between ebpfcore.sys and NMR.
The other is the version between the client and provider so this isn't the right version to bump.

Reply:
Reverted to 0.